cluster-autoscaler/utils/drain/drain_test.go

```diff
@@ -642,23 +642,30 @@ func TestDrain(t *testing.T) {
 			expectBlockingPod:   &BlockingPod{Pod: kubeSystemRcPod, Reason: UnmovableKubeSystemPod},
 			expectDaemonSetPods: []*apiv1.Pod{},
 		},
-		{
-			description:         "Custom-controller-managed blocking pod",
-			pods:                []*apiv1.Pod{customControllerPod},
-			pdbs:                []*policyv1.PodDisruptionBudget{},
-			expectFatal:         true,
-			expectPods:          []*apiv1.Pod{},
-			expectBlockingPod:   &BlockingPod{Pod: customControllerPod, Reason: NotReplicated},
-			expectDaemonSetPods: []*apiv1.Pod{},
-		},
 	}
 
-	for i := range tests {
-		// this flag is true by default in CA flags
-		tests[i].skipNodesWithCustomControllerPods = true
+	allTests := []testOpts{}
+	for _, sharedTest := range sharedTests {
+		// to execute the same shared tests for when the skipNodesWithCustomControllerPods flag is true
+		// and when the flag is false
+		sharedTest.skipNodesWithCustomControllerPods = true
```

Comment:
nit: You could update `sharedTest.description` to append a flag here, so it only affects the shared tests.

Reply:
Updated. :+1: